### DISCUSSION TITLE
Could not compile sample code due to ECMAScript compatibility issues.

### DISCUSSION THREAD

**Discussion by DavidSantacruzR** [aws-powertools/powertools-lambda-typescript#2788] [Q&A] [answered]
Hey, I'm just looking the aws powertools, specially the logger, and I ran into an issue when attempting to compile a project.

Steps to reproduce:

 * installing the logger `npm i @aws-lambda-powertools/logger`
* creating a sample.ts file with the following code:
`import { Logger } from '@aws-lambda-powertools/logger'; 
const logger = new Logger(); 
logger.info('Test');`
* Attempting to compile using `tsc sample.ts`

I am getting the error: Private identifiers are only available when targeting ECMAScript 2015 and higher.

node_modules/@aws-lambda-powertools/logger/lib/cjs/formatter/LogFormatter.d.ts:13
node_modules/@aws-lambda-powertools/logger/lib/cjs/Logger.d.ts

I have setup the target on the tsconfig to be "target": "ES2021" with no luck whatsoever.

Has anyone else experienced this issue?

**Comment by dreamorosi** [comment]
Hi @DavidSantacruzR, could you please copy the content of the `tsconfig.json` file here so I can test and see the error?

Thank you!

**Reply by dreamorosi** [reply, marked as answer]
I have tried to reproduce the issue you described and I was not able to.

Here you can see my setup: https://github.com/dreamorosi/ecmatest

**Reply by DavidSantacruzR** [reply]
Thank tou for your answer. This is my config file, and it differs quite a bit from yours.

I'll check my config based on what you've provided me with. Many many thanks.

{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "ES2021",
    "sourceMap": true,
    "esModuleInterop":true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": false,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false,
    "types": ["jest"]
  },
  "include": ["src", "__tests__"]
}
